docs/proposals/formatted-output.md

```diff
@@ -0,0 +1,312 @@
+
+For example, pull multiple files (layers) from a repository and filter out the file path of its first layer in the standard output. Then, pass the pulled first layer to the second command (`docker import`) to perform further operations. 
+
+```yaml
+jobs:
+  example-job:
+    steps:
+      - uses: oras-project/setup-oras@v1
+      - run: |
+          PATH=`oras pull $REGISTRY/$REPO:$TAG --format '{{.first .Files.Path}}'`
+          docker import $PATH
+```
+
+## Proposal and desired user experience
+
+Enable users to use the `--format <type>` flag to format output into structured data (e.g. JSON) and use the `--format` with the [Go template](https://pkg.go.dev/text/template) language to manipulate the output fields. Users can still use `--format '{{toRawJson .}}'` to get raw JSON output.
+
+### oras pull 
+
+When using `oras pull` with the flag `--format`, the following fields should be formatted into JSON output:
+
+- `Ref`: full artifact reference by digest, e.g, `$REGISTRY/$REPO@$DIGEST`
+-  `Files`: a list of downloaded files
+    -  `Path`: the absolute file path of the pulled file (layer)
+    -  `Ref`: full reference by digest of the pulled file (layer)
+    -  `MediaType`: media type of the pulled file (layer) 
+    -  `Digest`: digest of the pulled file (layer) 
+    -  `Size`: file size in bytes
+
+For example, pull an artifact that contains multiple layers (files) and show their descriptor metadata as pretty JSON in standard output:
+
+```bash
+oras pull $REGISTRY/$REPO:$TAG --artifact-type example/sbom sbom.spdx  --artifact-type example/vul-scan vul-scan.json --format json
+```
+
+```json
+{
+  "Ref": "localhost:5000/oras@sha256:7414904f07f515f48fe4afeaf876e3151039a81e7177b9c66e9e7ed6dd186111",
+  "Files": [
+    {
+      "Path": "/home/user/oras-install/sbom.spdx",
+      "Ref": "localhost:5000/oras@sha256:7414904f07f515f48fe4afeaf876e3151039a81e7177b9c66e9e7ed6dd186222",
+      "MediaType": "application/vnd.oci.image.manifest.v1+json",
+      "Digest": "sha256:7414904f07f515f48fe4afeaf876e3151039a81e7177b9c66e9e7ed6dd186222",
+      "Size": 820,
+      "Annotations": {
+        "org.opencontainers.image.title": "sbom.spdx"
+      }
+    },
+    {
+      "Path": "/home/user/oras-install/vul-scan.json",
+      "Ref": "localhost:5000/oras@sha256:7414904f07f515f48fe4afeaf876e3151039a81e7177b9c66e9e7ed6dd18669b",
+      "MediaType": "application/vnd.oci.image.manifest.v1+json",
+      "Digest": "sha256:7414904f07f515f48fe4afeaf876e3151039a81e7177b9c66e9e7ed6dd18669b",
+      "Size": 820,
+      "Annotations": {
+        "org.opencontainers.image.title": "vul-scan.json"
+      }
+    }
+  ]
+}
+```
+
+> [!NOTE]
+> When pulling a folder to filesystem, the value of `Path` should be an absolute path of the folder. Other fields are the same as the example of pulling files as above.
```

Comment:
How can user tell if a `Path` if a file or folder? Suggest to add `/` at the end of folder path.

Reply:
@qweeah That's a good point. I updated the Note to make it much strict and accurate.

When pulling a folder to filesystem, the value of `Path` should be an absolute path of the folder and should be end with slash `/` or backslash `\`, for example, `/home/Bob/sample-folder/` on Unix or `C:\Users\Bob\sample-folder\` on Windows. Other fields are the same as the example of pulling files as above.